lib/middleware/sidekiq/only_run_while_up.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Middleware
+  module Sidekiq
+    require 'down_time_checker'
+    class OnlyRunWhileUp
+      def call(worker, job, _queue)
+        if worker.respond_to?(:downtime_checks)
```

Comment:
Can you refactor this `call` code to something succinct?

Example:
```ruby
if should_do_downtime_checks? && services_down?
  reschedule_job
else
  yield
end
```

You might have to pass some vars there.